library/src/main/java/com/flipkart/okhttpstats/handler/PersistentStatsHandler.java

```diff
@@ -66,7 +72,8 @@
     private float mCurrentAvgSpeed = 0;
     private ConnectivityManager mConnectivityManager;
 
-    public PersistentStatsHandler(Context context) {
+    public PersistentStatsHandler(@NonNull Context context) {
+        this.mContext = context;
```

Comment:
What about storing the application context? `this.mContext = context.getApplicationContext()`?
Same on line 87.